examples/model_selection/plot_cost_sensitive_learning.py

```diff
@@ -0,0 +1,728 @@
+)
+print(
+    "Balanced accuracy of our intolerant classifier: "
+    f"{balanced_accuracy_scorer(intolerant_classifier, data_test, target_test):.3f}"
+)
+
+# %%
+# This is not a surprise that the balanced accuracy is at 0.5 for both classifiers.
+# However, we need to be careful in the rest of the evaluation: we potentially can
+# obtain a model with a decent balanced accuracy but that does not make any profit.
+# In this case, the model would be useless for our business.
+#
+# Let's now create a predictive model using a logistic regression without tuning the
+# decision threshold.
+from sklearn.linear_model import LogisticRegression
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+
+model = make_pipeline(StandardScaler(), LogisticRegression(random_state=42)).fit(
+    data_train, target_train
+)
+
+print(
+    "Benefit/cost of our logistic regression: "
+    f"${business_scorer(model, data_test, target_test, amount=amount_test):,.2f}"
+)
+print(
+    "Balanced accuracy of our logistic regression: "
+    f"{balanced_accuracy_scorer(model, data_test, target_test):.3f}"
+)
+
+# %%
+# By observing the balanced accuracy, we see that our predictive model is learning
+# some associations between the features and the target. The business metric also shows
+# that our model is beating the baseline in terms of profit and it would be already
+# beneficial to use it instead of ignoring the fraud detection problem.
+#
+# Now the question is: is our model optimum for the type of decision that we want to do?
+# Up to now, we did not optimize the decision threshold. We use the
+# :class:`~sklearn.model_selection.TunedThresholdClassifier` to optimize the decision
+# given our business scorer.
+tuned_model = TunedThresholdClassifier(
+    estimator=model,
+    objective_metric=business_scorer,
+    n_thresholds=100,
+    n_jobs=2,
+)
+
+# %%
+# Since our business scorer requires the amount of each transaction, we need to pass
+# this information in the `fit` method. The
+# :class:`~sklearn.model_selection.TunedThresholdClassifier` is in charge of
+# automatically dispatching this metadata to the underlying scorer.
+tuned_model.fit(data_train, target_train, amount=amount_train)
+
+# %%
+print(
+    "Benefit/cost of our logistic regression: "
+    f"${business_scorer(tuned_model, data_test, target_test, amount=amount_test):,.2f}"
+)
+print(
+    "Balanced accuracy of our logistic regression: "
+    f"{balanced_accuracy_scorer(tuned_model, data_test, target_test):.3f}"
+)
+
+# %%
+# We observe that tuning the decision threshold increases the profit of our model.
+# Eventually, the balanced accuracy also increased. Note that it might not always be
+# the case because the statistical metric is not necessarily a surrogate of the
+# business metric. It is therefore important, whenever possible, optimize the decision
+# threshold with respect to the business metric.
```

Comment:
```suggestion
# threshold with respect to the business metric.
#
# Finally, the estimate of the business metric itself can be unreliable, in
# particular when the number of data points in the minority class is so small.
# Any business impact estimated by cross-validation of a business metric on
# historical data (offline evaluation) should ideally be confirmed by A/B testing
# on live data (online evaluation). Note however that A/B testing models is
# beyond the scope of the scikit-learn library itself.
```

Reply:
Also, shall we show how to assess the variability of the business metric via cross validation?

```python
from sklearn.model_selection import cross_val_score
from sklearn.model_selection import StratifiedShuffleSplit

cross_val_score(
    tuned_model,
    data,
    target,
    params=dict(amount=amount),
    cv=StratifiedShuffleSplit(n_splits=5, test_size=0.5, random_state=42),
    scoring=business_scorer,
    n_jobs=2,
    verbose=10,
)
```

I find it interesting because it shows the power of routing side-metadata even in a cross-validation loop.

Reply:
Note that when I used `cv=5` I get very different numbers for the cost because it is not normalized w.r.t. to the test set size. Maybe we should normalize the business metric to output dollars per transaction instead of total dollars on the test set.